blog/content/post/2025-05-07-dasharo-ci.md

```diff
@@ -0,0 +1,293 @@
+  workflow_dispatch:
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v4
+      with:
+        path: edk2
+
+    - name: Checkout microsoft/secureboot_objects
+      uses: actions/checkout@v4
+      with:
+        repository: microsoft/secureboot_objects
+        path: secureboot_objects
+
+    - name: Check if DBX is out-of-date
+      run: |
+        old=$(sha256sum edk2/DasharoPayloadPkg/SecureBootDefaultKeys/DBXUpdate.bin | awk '{ print $1 }')
+        new=$(sha256sum secureboot_objects/PostSignedObjects/DBX/amd64/DBXUpdate.bin | awk '{ print $1 }')
+        if [ "$old" = "$new" ]; then
+          echo 'UEFI DBX is up-to-date.'
+        else
+          echo 'UEFI DBX is out of date.'
+          exit 1
+        fi
+
+  update:
+    runs-on: ubuntu-latest
+    needs: check
+    if: |
+      always() && needs.check.result == 'failure'
+
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v4
+      with:
+        path: edk2
+
+    - name: Checkout microsoft/secureboot_objects
+      uses: actions/checkout@v4
+      with:
+        repository: microsoft/secureboot_objects
+        path: secureboot_objects
+
+    - name: Update DBX blob
+      run: |
+        cp secureboot_objects/PostSignedObjects/DBX/amd64/DBXUpdate.bin edk2/DasharoPayloadPkg/SecureBootDefaultKeys/DBXUpdate.bin
+
+    - name: Set current date
+      run: |
+        pushd secureboot_objects
+        echo "RELEASE_DATE=$(git log -1 --pretty='format:%cs' PostSignedObjects/DBX/amd64/DBXUpdate.bin)" >> ${GITHUB_ENV}
+        popd
+
+    - name: Submit pull request
+      uses: peter-evans/create-pull-request@v7.0.7
+      with:
+        path: edk2
+        base: dasharo
+        branch: update_dbx_${{ env.RELEASE_DATE }}
+        title: Update DBX ${{ env.RELEASE_DATE }}
+        commit-message: "[automated change] Update DBX ${{ env.RELEASE_DATE }}"
+```
+
+We see the same overall logic:
+
+- Clone the secureboot objects repo
+- Make a checksum of the current dbx
+- Make a checksum of the latest dbx
+- Compare them
+- If the sums don't match:
+  - update the file
+  - create PR
+
+That's pretty much it. These two simple GitHub workflows automate updates of
+both microcode and the revocation database.
+
+## Closing thoughts
+
+Introducing these automatic checks makes our firmware not only more secure,
+but also more transparent. As the repositories and their CI workflows are open,
+each user can see for themselves when the microcode and DBX were last updated,
+and build from the main branch themselves with a guarantee that these components
+will be up to date each time.
+
+Other BIOS firmware vendors typically don't provide this information, and if
+they do, it's buried in the release notes. It's often not clear if the microcode
+is indeed the latest version available from the CPU vendor. We hope the
+introduction of these checks will make our firmware safer and more worthy of
+our users' trust.
```

Comment:
How our release notes are better from other vendors?

Reply:
Added https://github.com/3mdeb/news-and-ideas/pull/659/commits/382a588a9c03fc0e27e6e098d34e6975789c4455